src/object/cli_obj.c

```diff
@@ -4970,11 +4971,12 @@ obj_reasb_io_fini(struct obj_auxi_args *obj_auxi, bool retry)
 	if (retry && obj_auxi->reasb_req.orr_args != NULL) {
 		D_ASSERT(obj_auxi->reasb_req.orr_uiods != NULL);
 		obj_auxi->reasb_req.orr_args->iods = obj_auxi->reasb_req.orr_uiods;
-		obj_auxi->reasb_req.orr_args->sgls = obj_auxi->reasb_req.orr_usgls;
+		obj_auxi->reasb_req.orr_args->sgls = obj_auxi->reasb_req.orr_processed_sgls;
 	}
 	obj_bulk_fini(obj_auxi);
 	obj_auxi_free_failed_tgt_list(obj_auxi);
-	obj_dup_sgls_free(obj_auxi);
+	if (!retry)
```

Comment:
this line seems is the only real change, except the renaming?
TBH I'm rather keep the name of orr_usgls which means user input sgls, same as orr_uiods is user input iods.